app/models/invoice.rb

```diff
@@ -248,8 +249,50 @@ def void_invoice!
   def ensure_number
     return if number.present?
 
-    formatted_sequential_id = format('%03d', sequential_id)
+    if organization.document_numbering.to_s == 'per_customer'
+      customer_slug = "#{organization.document_number_prefix}-#{format('%03d', customer.sequential_id)}"
+      formatted_sequential_id = format('%03d', sequential_id)
 
-    self.number = "#{customer.slug}-#{formatted_sequential_id}"
+      self.number = "#{customer_slug}-#{formatted_sequential_id}"
+    else
+      org_formatted_sequential_id = format('%03d', organization_sequential_id)
+
+      self.number =
+        "#{organization.document_number_prefix}-#{Time.now.utc.strftime('%Y%m')}-#{org_formatted_sequential_id}"
```

Comment:
you can use some intermediate var to avoid this strange line return

Reply:
Added